planning/motion_velocity_planner/autoware_motion_velocity_run_out_module/src/parameters.hpp

```diff
@@ -0,0 +1,347 @@
+
+#ifndef PARAMETERS_HPP_
+#define PARAMETERS_HPP_
+
+#include <autoware/universe_utils/ros/parameter.hpp>
+#include <autoware/universe_utils/ros/update_param.hpp>
+#include <rclcpp/node.hpp>
+#include <rclcpp/parameter.hpp>
+
+#include <autoware_perception_msgs/msg/detail/object_classification__struct.hpp>
+#include <autoware_perception_msgs/msg/object_classification.hpp>
+
+#include <algorithm>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace autoware::motion_velocity_planner::run_out
+{
+/// @brief parameters for the filtering of the dynamic objects
+struct ObjectParameters
+{
+  bool ignore_if_stopped;
+  double stopped_velocity_threshold;
+  bool ignore_if_on_ego_trajectory;
+  bool ignore_if_behind_ego;
+  std::vector<std::string> ignore_objects_polygon_types;
+  std::vector<std::string> ignore_objects_lanelet_subtypes;
+  std::vector<std::string> ignore_collisions_polygon_types;
+  std::vector<std::string> ignore_collisions_lanelet_subtypes;
+  std::vector<std::string> cut_linestring_types;
+  std::vector<std::string> cut_polygon_types;
+  std::vector<std::string> cut_lanelet_subtypes;
+  bool cut_if_crossing_ego_from_behind;
+  double confidence_filtering_threshold;
+  bool confidence_filtering_only_use_highest;
+};
+/// @brief Parameters of the run_out module
+struct Parameters
+{
+  double
+    max_history_duration;  // [s]  calculated as the maximum duration among all buffer parameters
+  double stop_on_time_buffer;   // [s] successive collision detection time required to start the
+                                // stopping decision
+  double stop_off_time_buffer;  // [s] successive non-collision detection time required to remove a
+                                // stopping decision
+  double stop_distance_buffer;  // [m] longitudinal safety distance to keep between ego and the
+                                // collision position
+  double keep_stop_condition_time;  // [s] time along the ego trajectory where any collision will
+                                    // make us keep a stop decision
+  double keep_stop_condition_distance;  // [m] distance along the ego trajectory where any collision
+                                        // will make us keep a stop decision
+
+  double preventive_slowdown_on_time_buffer;  // [s] successive collision detection time required to
+                                              // start the slowdown decision
+  double preventive_slowdown_off_time_buffer;     // [s] successive non-collision detection time
+                                                  // required to remove the slowdown decision
+  double preventive_slowdown_distance_buffer;     // [m] longitudinal distance between the collision
+                                                  // and the slowdown positions
+  double preventive_slowdown_deceleration_limit;  // [m/s²] minimum deceleration that can be applied
+                                                  // by the preventive slowdown
```

Comment:
Nitpicking / personal preference

```suggestion
/// @brief Parameters of the run_out module
struct DecisionParameters
{
  double on_time_buffer{0.0};   // [s] continuous collision detection time required to activate decision
  double off_time_buffer{0.0};  // [s] continuous no-collision detection time required to activate decision
  double distance_buffer{0.0};  // [m] longitudinal safety distance to keep between ego and collision point
  double keep_time_th{0.0};     // [s] time along trajectory where any collision will keep the decision
  double keep_distance_th{0.0};     // [m] distance along trajectory where any collision will keep the decision
  double deceleration_limit{10.0};  // [m/s²] minimum deceleration that can be applied by the decision
};

struct Parameters
{
  double
    max_history_duration;  // [s]  calculated as the maximum duration among all buffer parameters

  DecisionParameters stop;  // parameters for the stop decision
  DecisionParameters slowdown;  // parameters for the preventive slowdown decision
```

Reply:
Thank you for the suggestion. I refactored the parameters a little but will do more refactoring in future PRs.